### DISCUSSION TITLE
WorkQueuePolicy - understand "disjoint filter subjects"

### DISCUSSION THREAD

**Discussion by Kilio22** [nats-io/nats-server#3637] [Q&A] [answered]
Hi,

I am new to NATS, and I am playing with Jetstream and the different retention policies, especially the WorkQueuePolicy.  
I noticed in the following example https://natsbyexample.com/examples/jetstream/workqueue-stream/go that you cannot create two consumers with overlaping filtrer subjects.

However, I manage to do the following using nats-cli and a local nats-server started in Jetstream mode:
* create a stream with `toto.>` as subject
* create a first pull consumer with `toto.*.tata` as subject
* create a second pull consumer with `toto.>` as subject

Here, to me the subjects are overlaping, as both can match a message published on a subject such as `toto.tutu.tata`.  
I was thinking that maybe it would result in receiving a message published on subject `toto.tutu.tata` only on the first consumer, but no, it is received by both consumers.

Could you explain why please?

Thanks!

**Comment by derekcollison** [comment, marked as answer]
WorkQueue based streams allow only one consumer, but you can attach as many applications as desired. You might look at InterestPolicy instead for your needs.

**Reply by derekcollison** [reply]
Actually I was wrong, you can have multiple as long as they are not overlapping. Seems we may have a bug, looking with Ivan now. Apologies!

**Reply by derekcollison** [reply]
Ivan found the issue and PR coming shortly and will be part of next release which is scheduled for tomorrow. Thanks for the report! Will let @kozlovic provide more info.

**Reply by Kilio22** [reply]
Glad to help!

**Reply by kozlovic** [reply]
I have created an issue (https://github.com/nats-io/nats-server/issues/3639) from this discussion and will have a PR shortly.
